File: dispatch/java/adamma/c4dhi/claid/Module/ChannelData.java
package adamma.c4dhi.claid.Module;

import java.time.LocalDateTime;

public class ChannelData<T>
{
    private final T data;
    private final LocalDateTime timestamp;
    private final String userToken;

    private final boolean valid;

    public ChannelData(final T data, final LocalDateTime timestamp, final String userToken)
    {
        this.data = data;
        this.timestamp = timestamp;
        this.userToken = userToken;
        this.valid = true;
    }

    public final T getData()
    {
        return this.data;
    }

    public final LocalDateTime getTimestamp()
    {
        return this.timestamp;
    }

    public final String getUserId()
    {
        return this.userToken;
    }

    public boolean isValid()
    {
        return this.valid;
    }

}

File: dispatch/python/module/channel_data.py
class ChannelData:

    def __init__(self, data = None, timestamp=None, user_id = None):
        self.data = data
        self.timestamp = timestamp
        self.user_id = user_id

        self.valid = False

        if self.data is not None and self.timestamp is not None and self.user_id is not None:
            self.valid = True

    def get_data(self):
        return self.data

    def get_timestamp(self):
        return self.timestamp

    def get_user_id(self):
        return self.user_id

    def is_valid(self):
        return self.valid